### PR TITLE
Refactored rdeckfacts definition to use facts_hash

### PR DESCRIPTION
**Pull request by abohne**
Fixes http://projects.theforeman.org/issues/1770

### CODE REVIEW

**Comment by ohadlevy** [comment]
[test]

File: app/models/host/managed.rb [hunk 1]
```diff
@@ -517,7 +517,10 @@ def set_ip_address
   def rundeck
     rdecktags = puppetclasses_names.map{|k| "class=#{k}"}
     unless self.params["rundeckfacts"].empty?
-      rdecktags += self.params["rundeckfacts"].split(",").map{|rdf| "#{rdf}=#{fact(rdf)[0].value}"}
+      rdecktags += self.params["rundeckfacts"].split(",").map { |rdf|
+        rdf.strip!
+        "#{rdf}=" + (facts_hash["{rdf}"] || "undefined")
```
Comment (domcleal) [inline app/models/host/managed.rb]: There's a `#` missing here, but you could just write: `facts_hash[rdf]`

**Comment by domcleal** [comment]
Thanks for submitting this @abohne, it looks good - just needs a slight change as above.  You can [squash this change](http://nathanhoad.net/git-amend-your-last-commit) into your current commit, and then overwrite the current branch with `git push -f origin` which will update this PR.

It's also good if the commit message references the bug ID, e.g.

```
fixes #1770 - refactored rdeckfacts definition to use facts_hash
```

(can be done with a commit amend)

File: app/models/host/managed.rb (hunk 1 above)
Comment (ohadlevy) [inline app/models/host/managed.rb]: @domcleal missing where? note its outside the quote

Comment (domcleal) [inline app/models/host/managed.rb]: @ohadlevy the facts_hash reference uses quotes around `{rdf}`

**Comment by abohne** [comment]
I've fixed that one typo (that's what I get for fixing on my production 1.1 instance :smile:)  and amended the commit.

I'll look into writing tests for this, but it might take a few days being that I'm only familiar enough with the code to hack this functionality back to a working state.

**Comment by domcleal** [comment]
I've added some tests in https://github.com/theforeman/foreman/pull/769.

**Pull request closed without merging**